src/core/Akka/Dispatch/Dispatchers.cs

```diff
@@ -143,40 +143,60 @@ internal sealed class ChannelExecutor : ExecutorService
         private Channel<IRunnable> _channel;
         private CancellationTokenSource _cts = new CancellationTokenSource();
 
-        private volatile int _readers = 0;
-        private bool HasReaders => _readers > 0;
+        private int _readers = 0;
+
+        public int DegreeOfParallelism { get; }
 
         public ChannelExecutor(string id, TaskScheduler scheduler, int degreeOfParallelism) : base(id)
         {
             _channel = Channel.CreateUnbounded<IRunnable>(new UnboundedChannelOptions()
             {
-                AllowSynchronousContinuations = true, SingleReader = false, SingleWriter = true
+                AllowSynchronousContinuations = true,
+                SingleReader = false,
+                SingleWriter = true
             });
 
-            for (var i = 0; i < degreeOfParallelism; i++)
-                Task.Run(() => ReadChannel(_channel.Reader));
+            DegreeOfParallelism = degreeOfParallelism;
         }
-        public static async Task ReadChannel(ChannelReader<IRunnable> reader)
+        public void ReadChannel()
         {
             try
             {
-                do
+                while (_channel.Reader.TryRead(out var runnable))
                 {
-                    while(reader.TryRead(out var runnable))
-                    {
-                        runnable.Run();
-                    }
-                } while (await reader.WaitToReadAsync().ConfigureAwait(false));
+                    runnable.Run();
+                }
             }
             catch
             {
                 // suppress exceptions
             }
+            finally
+            {
+                int initialValue, newValue;
+                do
+                {
+                    initialValue = _readers;
+                    newValue = _readers - 1;
+                } while (initialValue != Interlocked.CompareExchange(ref _readers, newValue, initialValue));
```

Comment:
`Interlocked.Decement(ref _readers)`?

Reply:
yeah that should probably be fine for this use case

Reply:
Made this change